app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -2809,16 +2809,27 @@ class BrowserTabViewModel @Inject constructor(
     fun onUserClickCtaSecondaryButton(cta: Cta) {
         viewModelScope.launch {
             ctaViewModel.onUserDismissedCta(cta)
-            if (cta is DaxBubbleCta.DaxPrivacyProCta) {
-                val updatedCta = refreshCta()
-                ctaViewState.value = currentCtaViewState().copy(cta = updatedCta)
-            } else if (cta is BrokenSitePromptDialogCta) {
+            if (cta is BrokenSitePromptDialogCta) {
                 onBrokenSiteCtaDismissButtonClicked(cta)
             }
-            if (cta is OnboardingDaxDialogCta.DaxFireButtonCta) {
-                pixel.fire(ONBOARDING_DAX_CTA_CANCEL_BUTTON, mapOf(PixelParameter.CTA_SHOWN to DAX_FIRE_DIALOG_CTA))
-                val updatedCta = ctaViewModel.getEndStaticDialogCta()
-                ctaViewState.value = currentCtaViewState().copy(cta = updatedCta)
+        }
+    }
+
+    fun onUserClickCtaDismissButton(cta: Cta) {
+        viewModelScope.launch {
+            onUserDismissedCta(cta)
+            if (cta is DaxBubbleCta) {
+                command.value = HideOnboardingDaxBubbleCta(cta)
+                pixel.fire(ONBOARDING_DAX_CTA_DISMISS_BUTTON, mapOf(PixelParameter.CTA_SHOWN to cta.ctaPixelParam))
+            } else if (cta is OnboardingDaxDialogCta) {
+                command.value = HideOnboardingDaxDialog(cta)
+                pixel.fire(ONBOARDING_DAX_CTA_DISMISS_BUTTON, mapOf(PixelParameter.CTA_SHOWN to cta.ctaPixelParam))
+                if (cta is OnboardingDaxDialogCta.DaxFireButtonCta) {
+                    val updatedCta = ctaViewModel.getEndStaticDialogCta()
+                    ctaViewState.value = currentCtaViewState().copy(cta = updatedCta)
```

Comment:
This is no longer needed. The expected behavior when a user dismisses the fire dialog is simply to close it and not show the end dialog anymore.